### ISSUE TITLE
Doesn't seem to work in default test config

### ISSUE THREAD

**Issue by jvsteiner** [continusec/verifiabledatastructures#1]
Hi,

I built it according to the README, and ran it in "demoMode".  The UI doesn't seem to contain a way to actually create a log or map, although there are two buttons to load an existing log or map.  using the load button configures the front end to mutate that log/map but makes no call to the server to create if it is missing.  trying to add an entry does not work, I get a 404 error for the log/map I am trying to mutate.

I hacked in some button to actually call the javascript create function via ```$("#createNewLog").click(function ()...```
But that doesn't work either, just get a 404, I thought due to account not being available, although the startup text suggest all are available.  So I tried to run in "realMode", but get same results, which is odd, because then the account info is explicitly configured in the config file.

**Comment by aeijdenberg** [comment]
Hi Jamie, thanks for the bug report (and please keep them going if this project is of interest to you). A couple of things:

1. This isn't ready for consumption yet :). As you may see from the commit history, the project is about one week old, and is the result of porting elements of a multi-tenant hosted offering into a simpler, more streamlined open-source version. I haven't actually sent the link anyone, and just flipped the public bit on GitHub on a whim, not actually expecting anyone to notice or see it, so I'd love to know how you found it!

2. There actually is no need to create a log or map before using it. Rather they should be lazily created on the first insert. Doing so in that manner makes reasoning on the API side about the life-cycle of the logs/maps simpler. The hosted version used to require creation of the objects before using them, so that's why some of the code still has references to that.

3. The bug you found is because yesterday I changed all of the data structures returned by the API to use protobuf generated classes, and also revved the API version on the server from v1 to v2. I hadn't updated the Javascript client yet to talk v2 API (it should mostly just be differing field names). I just made a commit to change the version number to v2 in the JS, but I haven't yet audited the code to verify that the operations work. I'll do so in the next day or so.

3. I'm still in the process of porting across some of my tests, and still need to update the client libraries (the Go one is now part of the server project, and ought to be serviceable, though the tests are broken and need updating) - one thing in particular that is on my list is an audit of the error codes. It's quite probable the error codes are currently misleading.

So, that means you should not be able to just insert an entry, though you'll need to "load" a log first (give it whatever name you want), but the other operations may not work yet..

Sorry this isn't polished yet - I'll update the README to reflect.

**Comment by aeijdenberg** [comment]
@jvsteiner, the Javascript client should now be up to date with the recent API changes, meaning that you ought to be able to click around in the demo console now without errors. Let me know if you find any more issues.

**Comment by jvsteiner** [comment]
looks like they are mostly fixed - I did run into a problem fetching a value from a map if it had not been inserted.  Expected result: proof of non-inclusion, Actual result was error: 
```
Uncaught DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.
    at RawDataEntryFactory.createFromLeafData (http://localhost:8092/continusec.js:1235:91)
    at http://localhost:8092/continusec.js:324:51
    at XMLHttpRequest.req.onload (http://localhost:8092/continusec.js:140:13)
```

Also, trying to find an entry that doesn't exist results in internal server error, and might possibly need to be not found or something else.  I think you mentioned above some of the error codes are misleading, so I know already you are probably thinking about this - thanks for the amazing responsiveness!

**Comment by aeijdenberg** [comment]
Thanks, should be fixed now in 9c5343463af651aacbff4cfa30c5c35c56b571b5.

I recently switched the REST API to use the same protobuf objects as used internally, marshaled as JSON instead of defining a mirrored set of JSON classes as it had previously done. An unfortunate side effect is that "empty" values are not emitted in JSON by the default protoc compiler - so unless I figure out how to change that, I need to update the JSON clients to add the empty values back in, which I've now done for this case.
